spark/interpreter/src/main/resources/python/zeppelin_pyspark.py

```diff
@@ -62,46 +62,33 @@ class PySparkCompletion:
   def __init__(self, interpreterObject):
     self.interpreterObject = interpreterObject
 
-  def getGlobalCompletion(self):
-    objectDefList = []
+  def getGlobalCompletion(self, text_value):
+    completions = [completion for completion in list(globals().keys()) if completion.startswith(text_value)]
+    return completions
+
+  def getMethodCompletion(self, objName, methodName):
+    execResult = locals()
     try:
-      for completionItem in list(globals().keys()):
-        objectDefList.append(completionItem)
+      exec("{} = dir({})".format("objectDefList", objName), globals(), execResult)
     except:
       return None
     else:
-      return objectDefList
+      objectDefList = execResult['objectDefList']
+      return [completion for completion in execResult['objectDefList'] if completion.startswith(methodName)]
 
-  def getMethodCompletion(self, text_value):
-    execResult = locals()
+  def getCompletion(self, text_value):
     if text_value == None:
       return None
-    completion_target = text_value
-    try:
-      if len(completion_target) <= 0:
-        return None
-      if text_value[-1] == ".":
-        completion_target = text_value[:-1]
-      exec("{} = dir({})".format("objectDefList", completion_target), globals(), execResult)
-    except:
-      return None
-    else:
-      return list(execResult['objectDefList'])
 
+    dotPos = text_value.find(".")
```

Comment:
be ware of the `..` https://stackoverflow.com/questions/43487811/what-is-python-dot-dot-notation-syntax

Reply:
Thanks @felixcheung , will do it in another PR since this PR is just trying improve the current PySparkInterpreter, not introducing new feature

Reply:
Can you/me open a JIRA to track?

Reply:
Sure, feel free to create ticket to track it , thanks

Reply:
done https://issues.apache.org/jira/browse/ZEPPELIN-3385